libs/platform/src/lib/components/form/base.input.ts

```diff
@@ -0,0 +1,267 @@
+import {
+    AfterViewInit,
+    ChangeDetectorRef,
+    DoCheck,
+    ElementRef,
+    Input,
+    OnChanges,
+    OnDestroy,
+    OnInit,
+    Optional,
+    Self,
+    SimpleChanges,
+    ViewChild
+} from '@angular/core';
+import { FormFieldControl } from './form-control';
+import { ControlValueAccessor, FormControl, NgControl, NgForm } from '@angular/forms';
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import { Subject } from 'rxjs';
+
+let randomId = 0;
+
+/**
+ * All form components share the same information (value, name, placeholder,.. ) as well as
+ * the same behavior given by ControlValueAccessor.
+ *
+ * Even this is not ideal solution there is no other way then use inheritance to reuse some of the
+ * common logic. It should be possible to use some kind of compositions with Proxies but something
+ * similar that exists in Aspect Oriented Programing.
+ *
+ */
+export abstract class BaseInput implements FormFieldControl<any>, ControlValueAccessor,
+    OnInit, OnChanges, DoCheck, AfterViewInit, OnDestroy {
+
+    protected defaultId: string = `fdp-id-${randomId++}`;
+
+    @Input()
+    id: string = this.defaultId;
+
+    @Input()
+    get name(): string {
+        return this._name ? this._name : this.id;
+    }
+
+    set name(value: string) {
+        this._name = value;
+        this.stateChanges.next('rb set name');
```

Comment:
Is the `name` property something we need to track changes on? Will the `name` of an input field change in an application or is there an operation that we need to do on setting of the input `name`?

Reply:
I have this only for debugging. But there are properties that are expected to change after is initialized and properties that never change. Next push is is going to reflect this.